File: Python/BT-FINETEST/BANTRON.py
def bantron(n, k):
    nums = list(range(1, n + 1))
    i = 0
    while len(nums) > 1:
        i = (i + k - 1) % len(nums)
        nums.pop(i)
    return nums[0]

n = int(input("So nguoi ngoi quanh ban: "))
k = 3
result = bantron(n, k)
print("Nguoi o lai cuoi cung la nguoi thu", result)

File: C++/ThucHanh/TH4_Mang/BANTRON.cpp
#include<iostream>
using namespace std;
void XoaPhanTu(int arr[],int &n,int index){
	for(int i=index;i<n-1;i++){
		arr[i]=arr[i+1];
	}
	n--;
}
int main(){
int n;
cout<<"So nguoi ngoi quanh ban: ";
cin>>n;
int *arr=new int[n];
for(int i=0;i<n;i++){
	arr[i]=i+1;
}
int index=0;
while(n>1){
	index=(index+2)%n;
	XoaPhanTu(arr,n,index);
}
cout<<"Nguoi o lai cuoi cung la nguoi thu "<<arr[0];
delete []arr;
return 0;
}

